ecs/pod_creator.go

```diff
@@ -27,11 +31,235 @@ func NewBasicECSPodCreator(c cocoa.ECSClient, v cocoa.Vault) (*BasicECSPodCreato
+		return nil, errors.Wrap(err, "invalid pod execution options")
+	}
+
+	secrets := m.getSecrets(mergedPodCreationOpts)
+
+	err := m.createSecrets(ctx, secrets)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating secrets")
+	}
+
+	taskDefinition, err := m.exportPodCreationOptions(ctx, mergedPodCreationOpts, secrets)
+	if err != nil {
+		return nil, errors.Wrap(err, "translating the register task definition input to the correct type")
+	}
+
+	registerOut, err := m.client.RegisterTaskDefinition(ctx, taskDefinition)
+	if err != nil {
+		return nil, errors.Wrap(err, "registering task definition")
+	}
+
+	if registerOut.TaskDefinition == nil || registerOut.TaskDefinition.TaskDefinitionArn == nil {
+		return nil, errors.New("expected a task definition from ECS, but none was returned")
+	}
+
+	taskDef := cocoa.NewECSTaskDefinition().SetID(*registerOut.TaskDefinition.TaskDefinitionArn).SetOwned(true)
+
+	runTask := m.exportTaskExecution(mergedPodExecutionOpts, *taskDef)
+
+	runOut, err := m.client.RunTask(ctx, runTask)
+	if err != nil {
+		return nil, errors.Wrap(err, "running task")
+	}
+
+	if runOut.Failures != nil {
+		catcher := grip.NewBasicCatcher()
+		for _, failure := range runOut.Failures {
+			catcher.Add(errors.Errorf("task '%s': %s: %s\n", *failure.Arn, *failure.Detail, *failure.Reason))
+		}
+		return nil, errors.Wrap(catcher.Resolve(), "running task")
+	}
+
+	if len(runOut.Tasks) == 0 || runOut.Tasks[0].TaskArn == nil {
+		return nil, errors.New("expected a task to be running in ECS, but none was returned")
+	}
+
+	resources := cocoa.NewECSPodResources().
+		SetCluster(*mergedPodExecutionOpts.Cluster).
+		SetSecrets(translatePodSecrets(secrets)).
+		SetTaskDefinition(*taskDef).
+		SetTaskID(*runOut.Tasks[0].TaskArn)
+
+	options := NewBasicECSPodOptions().
+		SetClient(m.client).
+		SetVault(m.vault).
+		SetStatus(cocoa.Running).
+		SetResources(*resources)
+
+	p, err := NewBasicECSPod(options)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "creating pod")
+	}
+
+	return p, nil
 }
 
 // CreatePodFromExistingDefinition creates a new pod backed by AWS ECS from an
 // existing definition.
 func (m *BasicECSPodCreator) CreatePodFromExistingDefinition(ctx context.Context, def cocoa.ECSTaskDefinition, opts ...*cocoa.ECSPodExecutionOptions) (cocoa.ECSPod, error) {
 	return nil, errors.New("TODO: implement")
 }
+
+// exportTags converts strings into ECS tags.
+func exportTags(tags []string) []*ecs.Tag {
+	ecsTags := []*ecs.Tag{}
+
+	for _, tag := range tags {
+		ecsTag := &ecs.Tag{}
+		ecsTag.SetKey(tag)
+		ecsTags = append(ecsTags, ecsTag)
+	}
+
+	return ecsTags
+}
+
+// exportStrategy converts the strategy and parameter into an ECS placement strategy.
+func exportStrategy(strategy *cocoa.ECSPlacementStrategy, param *cocoa.ECSStrategyParameter) []*ecs.PlacementStrategy {
+	placementStrat := ecs.PlacementStrategy{}
+
+	placementStrat.SetField(utility.FromStringPtr(param))
+
+	if strategy == nil {
```

Comment:
Hmm, looking back at how this code has changed, it seems like the original thing you did `placementStrat.SetType(string(*strategy))` might be best thing to do in thie case because if you do specify a placement strategy, you can't specify an empty strategy type. It might be easier to just lean on the fact that `Validate` will ensure that both type and field are set and just dereference it here.

Reply:
Sounds good, I'll change it back.